generic3g/specs/FrequencyAspect.F90

```diff
@@ -0,0 +1,140 @@
+#include "MAPL_Generic.h"
+module mapl3g_FrequencyAspect
+   use mapl3g_StateItemAspect
+   use mapl3g_AccumulatorActionInterface
+   use esmf
+   implicit none
+   private
+
+   public :: FrequencyAspect
+
+   type, extends(StateItemAspect) :: FrequencyAspect
+      type(ESMF_TimeInterval) :: dt
+      character(len=:), allocatable :: accumulation_type
+   contains
+      procedure :: matches
+      procedure :: supports_conversion_general
+      procedure :: supports_conversion_specific
+      procedure :: make_action
+   end type FrequencyAspect
+
+   interface FrequencyAspect
+      module procedure :: construct_frequency_aspect
+   end interface FrequencyAspect
+
+   interface operator(.divides.)
+      module procedure :: divides
+   end interface operator(.divides.)
+
+contains
+
+   function construct_frequency_aspect(dt, accumulation_type) result(aspect)
```

Comment:
We don't allow mirroring of frequency aspect.  But might be good to make it explicit in the constructor for clarity.

Reply:
Done